src/lib.rs

```diff
@@ -83,10 +140,55 @@ impl Sequence {
         (output >> 64) as u64
     }
 
-    fn shift_round(value: u64, n: u32) -> u64 {
-        (value + (1 << (n - 1))) >> n
-    }
+
+}
+
+fn shift_round(value: u64, n: u32) -> u64 {
+    (value + (1 << (n - 1))) >> n
 }
+
+/// Convert a raw sample bit pattern to an f32 in [0, 1).
+pub fn raw_to_f32(raw: u64) -> f32 {
+    (0.5 + f32::from_bits(0x3F80_0000 | shift_round(raw, 41) as u32)).fract()
+}
+
+/// Convert a raw sample bit pattern to an f64 in [0, 1).
+pub fn raw_to_f64(raw: u64) -> f64 {
+    (0.5 + f64::from_bits(0x3FF0_0000_0000_0000 | shift_round(raw, 12))).fract()
+}
+
+/// Convert a raw sample bit pattern to a 32-bit fixed-point number in [0, 1).
+pub fn raw_to_u32(raw: u64) -> u32 {
+    (1u32 << 31).wrapping_add(shift_round(raw, 32) as u32)
+}
+
+/// Convert a raw sample bit pattern to a 64-bit fixed-point number in [0, 1).
+pub fn raw_to_u64(raw: u64) -> u64 {
+    (1u64 << 63).wrapping_add(raw)
+}
+
+/// Scramble a raw sequence bit pattern with a scramble value and get the resulting f32 number in [0, 1).
+pub fn scramble_f32(raw: u64, scramble: u64) -> f32 {
```

Comment:
Did you find that scrambling actually works properly then? I did some highly unscientific tests visually inspecting some output and noticed that just XORing the raw 64-bit value doesn't preserve equidistribution

Reply:
Using addition instead will definitely work though since that's just adjusting the Rd "offset" parameter (currently hardcoded to 0.5 for symmetry reasons).

Reply:
Removed because it didn't work as wanted